designsafe/apps/rapid/opentopography_data/scripts/centerpoint_datagenerator.py

```diff
@@ -172,6 +187,28 @@ def save_aggregated_results(output_filepath, all_features):
     logging.info(f"Aggregated results saved to: {output_filepath}")
 
 
+def fetch_otcatalog_api_response(file_path):
+    with open(file_path, 'r') as file1:
+        data = json.load(file1)
+    features = data['features']
+    ot_data = fetch_otcatalog_data()
+    for feature in features:
+        id = ".".join(feature['properties']['id'].split('.')[1:])
+        dataset = ot_data.get(id)
+        if dataset is not None:
+            feature['properties']['description'] = dataset['description']
+            feature['properties']['doiUrl'] = dataset['url']
+            feature['properties']['host']='OpenTopo' if dataset['identifier']['propertyID']=='opentopoID' else dataset['identifier']['propertyID']
+            feature['properties']['dateCreated'] = dataset['dateCreated']
+            feature['properties']['temporalCoverage'] = dataset['temporalCoverage']
+            feature['properties']['keywords'] = dataset['keywords']
+        else:
+            logging.warning(f"Dataset with ID {feature['properties']['id']} not found in the OpenTopography API response.")
```

Comment:
Does this occur at the moment?

 If it does occur at the moment do, we need to remove it from the geojson response

If it doesn't occur, do we want to just throw an exception so that script stops and a developer is aware of the issue and can investigate